featurebyte/routes/common/schema.py

```diff
@@ -24,3 +26,7 @@
     default="_id", min_length=COLUMN_STR_MIN_LENGTH, max_length=COLUMN_STR_MAX_LENGTH
 )
 VerboseQuery = Query(default=False)
+
+
+# TODO: Change this to PyObjectId when upgrading to Pydantic v2
+PyObjectId = PydanticObjectId
```

Comment:
This should be `Change this to a PyObjectId class when upgrading to Pydantic V2`.

Reply:
Curious what is the problem with PydanticObjectid in upgrading to v2 and why do we need to define this alias?

Reply:
When pydantic upgrading to V2, we have different implementation for pydantic `ObjectId` used in route query parameter versus the one used in model. For the one used in route parameter, it behaves more like a `str`. For the one used in the model, it behaves more like a `ObjectId`. I did not find any single solution that fits both purpose. That's why the plan here is to split the current implementation into two different classes when we upgrade to pydantic V2.

Reply:
In pydantic V2, below are the implementations for `PyObjectId`:
```
class _ObjectIdPydanticAnnotation:
    # Based on https://docs.pydantic.dev/latest/usage/types/custom/#handling-third-party-types.

    @classmethod
    def __get_pydantic_core_schema__(
        cls,
        _source_type: Any,
        _handler: Callable[[Any], core_schema.CoreSchema],
    ) -> core_schema.CoreSchema:
        def validate_from_str(input_value: str) -> ObjectId:
            try:
                return ObjectId(input_value)
            except InvalidId:
                raise ValueError(f"Invalid ObjectId: {input_value}")

        return core_schema.union_schema(
            [
                # check if it's an instance first before doing any further work
                core_schema.is_instance_schema(ObjectId),
                core_schema.no_info_plain_validator_function(validate_from_str),
            ],
            serialization=core_schema.to_string_ser_schema(),
        )


PydanticObjectId = Annotated[ObjectId, _ObjectIdPydanticAnnotation]
```

`PyObjectId`
```
class PyObjectId(str):
    """
    Pydantic ObjectId type for API route path parameter
    """

    @classmethod
    def __get_pydantic_core_schema__(
        cls, _source_type: Any, _handler: Any
    ) -> core_schema.CoreSchema:
        return core_schema.json_or_python_schema(
            json_schema=core_schema.str_schema(),
            python_schema=core_schema.union_schema(
                [
                    core_schema.is_instance_schema(ObjectId),
                    core_schema.chain_schema(
                        [
                            core_schema.str_schema(),
                            core_schema.no_info_plain_validator_function(cls.validate),
                        ]
                    ),
                ]
            ),
            serialization=core_schema.plain_serializer_function_ser_schema(str),
        )

    @classmethod
    def validate(cls, value: Any) -> ObjectId:
        """
        Validate ObjectId
        Parameters
        ----------
        value: Any
            value to validate
        Returns
        -------
        ObjectId
            ObjectId value
        """
        if not ObjectId.is_valid(value):
            raise ValueError("Invalid ObjectId")
        return ObjectId(value)
```

Reply:
I see. We should use `PyObjectId` in route query parameters, and keep using `PydanticObjectId` in models, is that right?